### PR TITLE
Small TS style changes: Replace 'var' with 'let' or const as appropriate

### PR DESCRIPTION
**Pull request by agentydragon**
Also replaces some for-loops over arrays with for..of loops.

TS still compiles.

### CODE REVIEW

File: ts/src/editor.ts [hunk 1]
```diff
@@ -215,10 +214,10 @@ function onPaste(elem) {
 }
 
 function caretToEnd() {
-    var r = document.createRange();
+    let r = document.createRange();
```
Comment (dae) [inline ts/src/editor.ts]: You may already be aware of this, but const would be more appropriate in cases like this:  
https://mathiasbynens.be/notes/es6-const

File: ts/src/editor.ts [hunk 2]
```diff
@@ -494,20 +490,20 @@ allowedTagsExtended["SPAN"] = filterExternalSpan;
 Object.assign(allowedTagsExtended, allowedTagsBasic);
 
 // filtering from another field
-var filterInternalNode = function(node) {
+const filterInternalNode = function(node) {
```
Comment (dae) [inline ts/src/editor.ts]: It's probably better to use let for global functions, as there may be add-ons out there that are rebinding the functions to behave differently.

Comment (agentydragon) [reply ts/src/editor.ts]: Thanks, done

File: ts/src/editor.ts (hunk 1 above)
Comment (agentydragon) [reply ts/src/editor.ts]: Thanks, done

**Comment by dae** [comment]
Thanks!

**Pull request merged**